src/subscription.rs

```diff
@@ -199,8 +203,22 @@ impl<T: Runtime> EventStorageSubscription<T> {
     /// Gets the next change_set from the subscription.
     pub async fn next(&mut self) -> Option<StorageChangeSet<T::Hash>> {
         match self {
-            Self::Imported(event_sub) => event_sub.next().await,
+            Self::Imported(event_sub) => read_subscription_response(event_sub).await,
             Self::Finalized(event_sub) => event_sub.next().await,
         }
     }
 }
+
+async fn read_subscription_response<T>(sub: &mut Subscription<T>) -> Option<T>
+where
+    T: DeserializeOwned,
+{
+    match sub.next().await {
+        Ok(Some(next)) => Some(next),
+        Ok(None) => return None,
```

Comment:
Isn't `return` redundant here and below?

Reply:
nice catch, fixed